src/nvim/api/ui_events.in.h

```diff
@@ -68,4 +68,15 @@ void popupmenu_select(Integer selected)
 void tabline_update(Tabpage current, Array tabs)
     FUNC_API_SINCE(3) FUNC_API_REMOTE_ONLY;
 
+void cmdline_enter(void)
+    FUNC_API_SINCE(3) FUNC_API_REMOTE_ONLY;
+void cmdline_show(Array content, Integer pos, String firstc, String prompt, Integer level)
+    FUNC_API_SINCE(3) FUNC_API_REMOTE_ONLY;
+void cmdline_pos(Integer pos, Integer level)
+    FUNC_API_SINCE(3) FUNC_API_REMOTE_ONLY;
+void cmdline_char(String c, Integer shift)
```

Comment:
@justinmk What was the point of this suggestion? UI events are already prefectly extensible by adding new events, and new fields to extisting events. The only effect this has is that ext ui contributors would have to write a lot of dict packing code that the metadata mechanism could have autogenerated for them... The point of #6618 was exactly to _move away_ from `ADD("x", ...); ADD("y", ...); ui_event()` style code,  so I wonder why we want to cancel the effect of that PR by recreating `ui_event()` as an inner platform...